### ISSUE TITLE
Transitive Dependencies not showing in Dependency-Graph

### ISSUE THREAD

**Issue by almitte** [DependencyTrack/dependency-track#1513]
### Current Behavior:
Transitive dependencies where the child node (second hierarchy level) is inside the "components" of the parent node (first hierarchy level) are not being displayed in Dependency-Graph. Error was noticed after using hierarchical Merge with CycloneDX-Cli, as only the first hierarchy level (the primary-components of the merged SBoMs) are showing in Dependency-Graph in Dtrack, but no second hierarchy level components.
Not working Code:
![grafik(4)](https://user-images.githubusercontent.com/101641863/161799568-67ed6a94-5605-4f88-bed4-102dd20f6652.png)
Result:
![2](https://user-images.githubusercontent.com/101641863/161799655-e3f9471c-068c-415c-9e57-4a9002c63540.png)

Working Code:
![grafik(5)](https://user-images.githubusercontent.com/101641863/161801168-f7e24a8b-b5f1-46db-8371-3b8aadae33ae.png)
Result:
![grafik(2)](https://user-images.githubusercontent.com/101641863/161799749-dd4098f7-fb0a-4799-be4d-b1087f3e9183.png)

Dependencies:
![grafik(6)](https://user-images.githubusercontent.com/101641863/161801218-8244cb86-a48f-45f4-bb49-e72072025f32.png)

### Steps to Reproduce:
Add a child node to "components" of a first hierarchy level parent node and add this child node to "Dependencies" as well.
Also merge two valid SBoMs using the CycloneDX-Cli with the `--hierarchical` Setting. (Resulting SBoMs Dependency-Graph in Dtrack only displays first hierarchy level components.)

### Expected Behavior:
Dependency-Graph should display transitive dependencies.

### Environment:
  
  - Dependency-Track Version: latest
  - Distribution: Docker
  - BOM Format & Version: JSON, 1.4
  - Database Server: SQL-Server

### Additional Details:

(e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

**Comment by stevespringett** [comment]
This is a frontend issue.

Duplicate of https://github.com/DependencyTrack/frontend/issues/85

**Comment by roadSurfer** [comment]
I am a bit confused about if this should this be fixed or not. It's marked as `duplicate`, https://github.com/DependencyTrack/frontend/issues/85 is closed, but I am hitting the issue in 4.7.0 after running a hierarchical merge.

**Comment by roadSurfer** [comment]
To illustrate what I am seeing, I created two simple BOMs for `foo` and `bar`, and uploaded those to DT. Both of these displayed fine:
![foo_depgraph](https://user-images.githubusercontent.com/38717523/214806625-d5946b2c-0e0c-477c-9b50-9b183cfe0c08.png)
![bar_depgraph](https://user-images.githubusercontent.com/38717523/214806642-6cedbc8c-31d8-43fd-95e6-e9dadfe96f49.png)

I then ran a hierarchical merge to create a BOM for `foobar`: `docker run --rm -v <bom_location>:/tmp/target --name cyclonedx cyclonedx/cyclonedx-cli:0.24.2 merge --input-files /tmp/target/bom_foo.xml /tmp/target/bom_bar.xml --input-format xml --output-file /tmp/target/bom_foobar.xml --output-format xml --hierarchical --group tld.domain --name foobar --version 0.0.1-SNAPSHOT`
Uploading that to DT shows the problem (although the BOM itself seems fine):
![foobar_depgraph](https://user-images.githubusercontent.com/38717523/214806983-98e92729-2032-4e20-b032-402f6123d8ef.png)
Note how there is no expansion of the `foo` and `bar` dependencies when expected.

[boms.zip](https://github.com/DependencyTrack/dependency-track/files/10508105/boms.zip)

**Comment by roadSurfer** [comment]
Looking in the DB (postgres in this case), I can see that `direct_dependencies` is empty for `foo`. This to me implies that the fault lies somewhere withing the ingestion of the BOM.
![missing_directs](https://user-images.githubusercontent.com/38717523/214834364-b6aa876c-dfe2-40fd-b535-19f21f20dd38.png)

**Comment by stevespringett** [comment]
@almitte Closing this issue as the navigation down to transitive dependencies was resolved in v4.7.

@roadSurfer the issue you're experiencing is due to both of your direct dependencies not having any dependences of their own.

Neither of these dependencies have any dependencies defined in the BOM
```
tld.domain.foo@0.0.1-SNAPSHOT:pkg:maven/tld.domain/foo@0.0.1-SNAPSHOT?type=jar
tld.domain.bar@0.0.1-SNAPSHOT:pkg:maven/tld.domain/bar@0.0.1-SNAPSHOT?type=jar
```
